core/iwasm/common/wasm_c_api.c

```diff
@@ -418,37 +425,32 @@ static korp_mutex engine_lock = OS_THREAD_MUTEX_INITIALIZER;
 #endif
 
 own wasm_engine_t *
-wasm_engine_new_with_args(mem_alloc_type_t type, const MemAllocOption *opts)
```

Comment:
This is an ABI change, not sure whether to remove it? If yes, we need to add it in the "Breaking changes" of the [release notes](https://github.com/bytecodealliance/wasm-micro-runtime/pull/2821/files).

Reply:
You are right. I am gona keep the old function and use a `deprecated` attribute to highlight.